### PR TITLE
Fix issue where an invalid URL using the Client would crash

### LINKED ISSUE #2716: Crash when trying to perform a request with invalid URL using client
### Describe the bug

The process crashes when performing a request with `URI` that `URL(string: uri.absoluteString)` evaluates as `nil`.

### To Reproduce

call 
```swift
req.client.get("https://api.github.com/repos/ColdGrub1384/Pisth Terminal")
```

vapor crashes on line https://github.com/vapor/vapor/blob/main/Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift#L21 because `URL(string: client.url.string)!` is evaluated as `nil` and force-unwrapped.

### Expected behavior

An error should be raised instead of crashing

### Environment

* Vapor Framework version: 4.48.0 (4.49.2 still seem to contain this bug based on the source code)
* Vapor Toolbox version: irrelevant
* OS version: irrelevant

### PR DESCRIPTION
**Pull request by NguyenQuyThien**
This resolves an issue where sending a request with an invalid URL using the Client cause a crash. 
Fixes #2716.

### CODE REVIEW

File: Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift [hunk 1]
```diff
@@ -16,9 +16,13 @@ private struct EventLoopHTTPClient: Client {
     func send(
         _ client: ClientRequest
     ) -> EventLoopFuture<ClientResponse> {
+        guard let url = URL(string: client.url.string) else {
+            self.logger?.error("\"\(client.url.string)\" is an invalid URL")
```
[the remark this answers is not in the document]
Comment (NguyenQuyThien) [reply Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift:20]: I assume you are referring to the logging type. It should be `self.logger?.debug("\"\(client.url.string)\" is an invalid URL")` Right?

File: Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift [hunk 2]
```diff
@@ -16,9 +16,13 @@ private struct EventLoopHTTPClient: Client {
...
+            return self.eventLoop.makeSucceededFuture(ClientResponse())
```
[the remark this answers is not in the document]
Comment (NguyenQuyThien) [reply Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift:21]: I got it! I will research the source code.

File: Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xTim) [reply Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift:20]: Correct yep!

File: Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xTim) [reply Sources/Vapor/HTTP/Client/EventLoopHTTPClient.swift:21]: You can do
```suggestion
            return self.eventLoop.future(error: Abort(.internalServerError))
```

**Review by 0xTim — APPROVED** [review]
Thanks!

**Pull request merged**